### PR TITLE
Fix #125 - Add BBox support also for Position

### LINKED ISSUE #125: Add BBox support also for Position

### PR DESCRIPTION
**Pull request by flyusfly**

### CODE REVIEW

File: ban/http/resources.py [hunk 1]
```diff
@@ -173,6 +173,32 @@ class Position(VersionnedResource):
...
+    def get_collection(self, req, resp, **kwargs):
+        qs = super().get_collection(req, resp, **kwargs)
+        bbox = self.get_bbox(req)
+        if bbox:
+            qs = (qs.where(models.Position.center.in_bbox(**bbox))
+                    .group_by(models.Position.pk)
+                    .order_by(models.Position.pk))
+        return qs
+
+    @app.endpoint('/{identifier}/positions')
```
Comment (yohanboniface) [inline ban/http/resources.py]: Why do we have this new method? Asking "positions" on Position resource does not seem that natural to me ;)

File: ban/http/resources.py [hunk 2]
```diff
@@ -173,6 +173,32 @@ class Position(VersionnedResource):
     """Manipulate position resources."""
     model = models.Position
 
+    def get_bbox(self, req):
```
Comment (yohanboniface) [inline ban/http/resources.py]: This needs to be factorized with HouseNumber class.

File: ban/http/resources.py [hunk 3]
```diff
@@ -173,6 +173,32 @@ class Position(VersionnedResource):
...
+        if not len(bbox) == 4:
+            return None
+        return bbox
+
+    def get_collection(self, req, resp, **kwargs):
+        qs = super().get_collection(req, resp, **kwargs)
+        bbox = self.get_bbox(req)
+        if bbox:
+            qs = (qs.where(models.Position.center.in_bbox(**bbox))
+                    .group_by(models.Position.pk)
```
Comment (yohanboniface) [inline ban/http/resources.py]: No need for GROUP BY here, given we are not doing any join.

**Comment by yohanboniface** [comment]
Can you rebase on master (or I can do it if you prefer)? the diff is somehow messy :s

**Comment by flyusfly** [comment]
I will rebase it!
Thx in advance for your feedback.

**Comment by flyusfly** [comment]
No more conflicts with master branch.
Is it OK for you to merge it?

File: ban/http/resources.py [hunk 4]
```diff
@@ -197,6 +190,32 @@ def get_bbox(self, req):
...
+
+class Position(VersionnedResource, BboxResource):
+    """Manipulate position resources."""
+    model = models.Position
+
+    def get_collection(self, req, resp, **kwargs):
+        qs = super().get_collection(req, resp, **kwargs)
+        bbox = self.get_bbox(req)
+        if bbox:
+            qs = (qs.where(models.Position.center.in_bbox(**bbox)))
```
Comment (yohanboniface) [inline ban/http/resources.py]: No need for the parens

**Pull request merged**